### PR TITLE
mmc: add required delays when retrying commands

### PR DESCRIPTION
**Pull request by Yann-lms**
A new function mmc_reset_to_idle is also created.

Signed-off-by: Yann Gautier <yann.gautier@st.com>

### CODE REVIEW

**Comment by antonio-nino-diaz-arm** [comment]
IP Review +1

File: drivers/mmc/mmc.c [hunk 1]
```diff
@@ -385,6 +401,8 @@ static int mmc_send_op_cond(void)
 			return ret;
 		}
 
+		udelay(100);
+
 		retries--;
 	} while ((resp_data[0] & OCR_POWERUP) == 0U);
```
[the remark this answers is not in the document]
Comment (Yann-lms) [reply drivers/mmc/mmc.c]: OK, and that looks easier to read. I may change the other retry loops like this as well.

**Comment by Yann-lms** [comment]
I've updated the patch with your suggestion, but changed:
` if ((resp_data[0] & OCR_POWERUP) == 0U)`
to
` if ((resp_data[0] & OCR_POWERUP) != 0U)`

**Comment by Yann-lms** [comment]
If PR #1504 is merged before this one, there may be merge conflicts. The new patch can be found there:
https://github.com/Yann-lms/arm-trusted-firmware/commit/0c7f8f1c18fbdec3ee124316515b7b70a10f24b5

**Pull request merged**